src/rebar_prv_shell.erl

```diff
@@ -129,3 +131,45 @@ add_test_paths(State) ->
                   end, rebar_state:project_apps(State)),
     _ = code:add_path(filename:join([rebar_dir:base_dir(State), "test"])),
     ok.
+
+reread_config(State) ->
+    ConfigList = find_config(State),
+    lists:foreach(fun ({Application, Items}) ->
+                          lists:foreach(fun ({Key, Val}) ->
+                                                application:set_env(Application, Key, Val)
+                                        end,
+                                        Items)
+                  end,
+                  ConfigList).
+
+% First try the --config flag, then try the relx sys_config
+-spec find_config(rebar_state:t()) -> [tuple()].
+find_config(State) ->
+    case find_config_option(State) of
+        [] ->
+            find_config_relx(State);
```

Comment:
This is technically wrong. If I ever pass a config explicitly as `--config=myfile` and `myfile` contains nothing but empty lines, I should get `[]` as a final config; however because this function matches on "empty or not empty", and empty config file may just mean that relx will be used instead.

`find_config/1` should have a return type of `{ok, List} | not_found` or something similar that lets us properly know if there is an empty config or no config at all that has been passed.